### DISCUSSION TITLE
How can I prevent sending a command that is pre-filled by the cisco device in the CLI?

### DISCUSSION THREAD

**Discussion by dowiz** [ktbyers/netmiko#3423] [Q&A]
Hello! I have a task to get the available features from a cisco device. 
I found a way to get them using a command with a question mark at the end. 
But I have a problem that occurs periodically, for example, when sending a request:

```
device = {
        'device_type': 'cisco_ios',
        'ip':   ip,
        'port': port,
        'username': username,
        'password': password,
        'secret': password,
    }

connection = ConnectHandler(**device)

connection.enable()

info = connection.send_command("show vlan ?")

connection.disconnect()

print(info)

```

I get the responses (either 1 or 2):
1:
>   brief        VTP all VLAN status in brief
>   group        VLAN group(s) information
>   id           VTP VLAN status by VLAN id
>   ifindex      SNMP ifIndex
>   internal     VLAN internal usage
>   mtu          VLAN MTU information
>   name         VTP VLAN status by VLAN name
>   remote-span  Remote SPAN VLANs
>   summary      VLAN summary information
>   |            Output modifiers
>   <cr>
> 
> sw2.school#show vlan
> 

2:
>   brief        VTP all VLAN status in brief
>   group        VLAN group(s) information
>   id           VTP VLAN status by VLAN id
>   ifindex      SNMP ifIndex
>   internal     VLAN internal usage
>   mtu          VLAN MTU information
>   name         VTP VLAN status by VLAN name
>   remote-span  Remote SPAN VLANs
>   summary      VLAN summary information
>   |            Output modifiers
>   <cr>
> 
> sw2.school#show vlan
> 
> VLAN Name                             Status    Ports
> ---- -------------------------------- --------- -------------------------------
> 1    default                          active    Fa0/1, Fa0/2, Fa0/3, Fa0/4, Fa0/5, Fa0/6, Fa0/7, Fa0/8, Fa0/9, Fa0/10, Fa0/11, Fa0/12, Fa0/13, Fa0/14, Fa0/15, Fa0/16, Fa0/17, Fa0/18, Fa0/20
> 5    Wi-Fi-School                     active    Fa0/19, Fa0/21, Fa0/22, Fa0/23, Fa0/24
> 1002 fddi-default                     act/unsup
> 1003 token-ring-default               act/unsup
> 1004 fddinet-default                  act/unsup
> 1005 trnet-default                    act/unsup
> 
> VLAN Type  SAID       MTU   Parent RingNo BridgeNo Stp  BrdgMode Trans1 Trans2
> ---- ----- ---------- ----- ------ ------ -------- ---- -------- ------ ------
> 1    enet  100001     1500  -      -      -        -    -        0      0
> 5    enet  100005     1500  -      -      -        -    -        0      0
> 1002 fddi  101002     1500  -      -      -        -    -        0      0
> 1003 tr    101003     1500  -      -      -        -    -        0      0
> 1004 fdnet 101004     1500  -      -      -        ieee -        0      0
> 1005 trnet 101005     1500  -      -      -        ibm  -        0      0
> 
> Remote SPAN VLANs
> ------------------------------------------------------------------------------
> 
> 
> Primary Secondary Type              Ports
> ------- --------- ----------------- ------------------------------------------
> 

The appearance of the second answer is very dangerous, if I don't start such a query with "show", I can get into serious trouble!

How can I prevent sending a command that is pre-filled by the device in the CLI?

**Comment by ktbyers** [comment]
You should use `write_channel` and `read_channel`, using the `?` the way you are is not what `send_command` is intended for.

Another option is possibly to use `send_command`, but add the `expect_string` argument to indicate the terminating pattern. This might also fix your issue.
